contrib/babelfishpg_tsql/src/hooks.c

```diff
@@ -563,6 +588,188 @@ pltsql_GetNewObjectId(VariableCache variableCache)
 	ShmemVariableCache->oidCount = 0;
 }
 
+static Oid
+pltsql_GetNewTempObjectId()
+{
+	Oid			result;
+	Oid			tempOidStart;
+	static Oid 	nextTempOid = InvalidOid;
+
+	/* safety check, we should never get this far in a HS standby */
+	if (RecoveryInProgress())
+		elog(ERROR, "cannot assign OIDs during recovery");
+
+	/*
+	 * temp_oid_buffer_size = 0 would indicate that the feature is 
+	 * disabled, so we shouldn't even reach this code.
+	 */
+	Assert(temp_oid_buffer_size > 0);
```

Comment:
FATAL might be too aggressive. I would leave it at ERROR, enough to stop the current query. I don't see a reason to kill the entire server because of this.